lib/electrode-server.js

```diff
@@ -210,16 +210,12 @@ async function startElectrodeServer(context) {
     // -    3rd call wait for all registration to complete
     //
     const regPromises = plugins.map(plugin => {
-      try {
-        const x = server.register(plugin.register, plugin.options);
-        return util
-          .promisify(x.after)()
-          .catch(err => {
-            throw regFail(err, plugin);
-          });
-      } catch (err) {
-        return Promise.reject(regFail(err, plugin));
-      }
+      const x = server.register(plugin.register, plugin.options);
+      return util
+        .promisify(x.after)()
+        .catch(err => {
+          throw regFail(err, plugin);
+        });
```

Comment:
If server.register throws a sync error then it will not be caught and go through `regFail`.

do this:

```js
async plugin => {
        try {
          const x = server.register(plugin.register, plugin.options)
          return await util.promisify(x.after)();
        } catch (err) {
          throw regFail(err, plugin)
        }
      }
```